Centaurus.Domain/WebSockets/Auditor/AuditorWebSocketConnection.cs

```diff
@@ -12,24 +12,63 @@
 
 namespace Centaurus.Domain
 {
-    public class AuditorWebSocketConnection : ClientWebSocketConnection
+    public class AuditorWebSocketConnection : BaseWebSocketConnection
     {
 
         static Logger logger = LogManager.GetCurrentClassLogger();
 
         CancellationTokenSource cancellationToken = new CancellationTokenSource();
 
-        protected override async Task<bool> HandleMessage(MessageEnvelope envelope)
+        public AuditorWebSocketConnection(WebSocket webSocket)
+            : base(webSocket)
         {
-            return await MessageHandlers<AuditorWebSocketConnection>.HandleMessage(this, envelope);
+            //we don't need to create and sign heartbeat message on every sending
+            hearbeatMessage = new Heartbeat().CreateEnvelope();
+            hearbeatMessage.Sign(Global.Settings.KeyPair);
+#if !DEBUG
+            InitTimer();
+#endif
+        }
+
+        private System.Timers.Timer heartbeatTimer = null;
+
+        //If we didn't receive message during specified interval, we should close connection
+        private void InitTimer()
+        {
+            heartbeatTimer = new System.Timers.Timer();
+            heartbeatTimer.Interval = 5000;
+            heartbeatTimer.AutoReset = false;
+            heartbeatTimer.Elapsed += HeartbeatTimer_Elapsed;
         }
 
-        public override async Task EstablishConnection()
+        private MessageEnvelope hearbeatMessage;
+
+        private async void HeartbeatTimer_Elapsed(object sender, System.Timers.ElapsedEventArgs e)
         {
-            await base.EstablishConnection();
+            await SendMessage(hearbeatMessage);
+        }
+
+        protected ClientWebSocket clientWebSocket => webSocket as ClientWebSocket;
```

Comment:
Duplicate code. Move duplicate code to the base class.

Reply:
This code is duplicated only in tests, so it makes no sense to put it in the base class